pkg/apiserver/watch.go

```diff
@@ -181,11 +182,20 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	flusher.Flush()
 
+	delay := wait.NewChannelDelayedAction(
+		wait.Interval{Count: 0, Interval: 5 * time.Millisecond},
+		wait.Interval{Count: 10, Interval: 100 * time.Millisecond},
+		wait.Interval{Count: 100, Interval: 500 * time.Millisecond},
+	)
+
 	buf := &bytes.Buffer{}
 	for {
 		select {
 		case <-cn.CloseNotify():
 			return
+		case <-delay.After():
+			delay.Done()
+			flusher.Flush()
```

Comment:
Good point

On Tue, Apr 26, 2016 at 7:25 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/apiserver/watch.go
> <https://github.com/kubernetes/kubernetes/pull/24768#discussion_r61068898>
> :
>
> >  	buf := &bytes.Buffer{}
> >  	for {
> >  		select {
> >  		case <-cn.CloseNotify():
> >  			return
> > +		case <-delay.After():
> > +			delay.Done()
> > +			flusher.Flush()
>
> We should probably also flush at the end (e.g. if it finishes because of
> timeout).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24768/files/91a0c160abd3ac01d0dbe9a89b3b5131c0cad419#r61068898>
>